packages/dataclass-editor/src/DataClassEditor.tsx

```diff
@@ -1,5 +1,46 @@
-function DataClassEditor() {
-  return 'Data Class Editor';
+import { Flex, PanelMessage, Spinner } from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect, useMemo, useState } from 'react';
+import './DataClassEditor.css';
+import { useClient } from './protocol/ClientContextProvider';
+import type { EditorProps } from './protocol/types';
+import { genQueryKey } from './query/query-client';
+
+function DataClassEditor(props: EditorProps) {
+  const [context, setContext] = useState(props.context);
+  useEffect(() => {
+    setContext(props.context);
+  }, [props]);
+
+  const client = useClient();
+
+  const queryKeys = useMemo(() => {
+    return {
+      data: () => genQueryKey('data', context),
+      saveData: () => genQueryKey('saveData', context)
+    };
+  }, [context]);
+
+  const { data, isPending, isError, error } = useQuery({
+    queryKey: queryKeys.data(),
+    queryFn: () => client.data(context),
+    structuralSharing: false
+  });
+
+  if (isPending) {
+    return (
+      <Flex alignItems='center' justifyContent='center' style={{ width: '100%', height: '100%' }}>
+        <Spinner />
+      </Flex>
+    );
+  }
+
+  if (isError) {
+    return <PanelMessage icon={IvyIcons.ErrorXMark} message={`An error has occurred: ${error.message}`} />;
```

Comment:
maybe you need to wrap this too in a container which is 100% height? you can easily test it over the tanstack query devtools, when you trigger an error ;)

Reply:
Looks good. No need for it.